File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int z = 0;
        int n = nums.size();
        int t = n-1;
        int curr = 0;
        while(curr<=t){
            if(nums[curr]==0){
                swap(nums[curr],nums[z]);
                curr++;
                z++;
            }
            else if(nums[curr]==1){
                curr++;
            }
            else{
                swap(nums[curr],nums[t]);
                t--;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int z = 0;
        
        int n = nums.length;
        int t = n-1;
        int curr = 0;
        while(curr<=t){
            if(nums[curr]==1){
                curr++;
                continue;
            }
            else if(nums[curr]==0){
                nums[curr]=nums[z];
                nums[z]=0;
                z++;
                curr++;
            }
            else if (nums[curr]==2){
                nums[curr]=nums[t];
                nums[t]=2;
                t--;
            }
        }
    }
}